### PR TITLE
Change Quickstart to Use Tabular Data

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I adjusted the quickstart to use a tabular dataset (iris flower classification).

Other changes:
- I modified FacetsVisualizer to be able to directly take a dict of outputs / output artifacts as arguments such that we no longer need a separate step in the pipeline if we already have pandas artifacts somewhere. This means we can remove the `skew_comparison()` steps from both training and inference pipelines in the quickstart.
- I removed lot's of dead/useless code: 
  - `zenml.integration.sklearn.helpers`, 
  - built-in pipeplines, 
  - built-in steps, 
  - sklearn standard steps,
  - tensorflow standard steps,
  - unused step interfaces, 
  - `BaseVisualizer` subclasses.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

**Review by strickvl — APPROVED** [review]
Looks good to me. Only the 50% holdout for the test set was something I wasn't sure about.

**Review by AlexejPenner — APPROVED** [review]
Nothing to add to Alex's comments - great job :broom:

**Review by bcdurak — APPROVED** [review]
LGTM!

File: docs/book/developer-guide/steps-pipelines/pipeline-visualization.md [hunk 1]
```diff
@@ -84,11 +85,15 @@ from zenml.repository import Repository
...
+def digits_data_loader() -> Output(
     X_train=np.ndarray, X_test=np.ndarray, y_train=np.ndarray, y_test=np.ndarray
 ):
-    """Loads the digits dataset as normal numpy arrays."""
-    X_train, X_test, y_train, y_test = get_digits()
+    """Loads the digits dataset as a tuple of flattened numpy arrays."""
+    digits = load_digits()
+    data = digits.images.reshape((len(digits.images), -1))
+    X_train, X_test, y_train, y_test = train_test_split(
+        data, digits.target, test_size=0.5, shuffle=False
```
[the remark this answers is not in the document]
Comment (fa9r) [reply docs/book/developer-guide/steps-pipelines/pipeline-visualization.md:95]: Yeah 50% is quite a lot, I changed it to 20%.

File: examples/quickstart/README.md [hunk 2]
```diff
@@ -8,10 +8,10 @@ Then open [notebooks/quickstart.ipynb](notebooks/quickstart.ipynb) to get starte
...
 
 This quickstart aims to give you a small illustration of what ZenML can do. To do so, we will:
-- Train a model on the MNIST digits dataset, evaluate it, deploy it, and embed it in an inference pipeline
-- Automatically version, track, and cache data, models, and other artifacts
-- Track model hyperparameters and metrics in an experiment tracking tool
-- Measure and visualize train-test skew, training-serving skew, and data drift
+- Train a model on the iris flower classification dataset, evaluate it, deploy it, and embed it in an inference pipeline,
+- Automatically version, track, and cache data, models, and other artifacts,
+- Track model hyperparameters and metrics in an experiment tracking tool,
+- Measure and visualize train-test skew, training-serving skew, and data drift.
```
[the remark this answers is not in the document]
Comment (fa9r) [reply examples/quickstart/README.md:14]: Isn't this the grammatically correct way to format lists though?

[the remark this answers is not in the document]
Comment (strickvl) [reply examples/quickstart/README.md:14]: Technically, maybe, but only if you consider lists to be part of a whole long prose text.

I think it's also really common just to see lists without commas in between (i.e. because the bullet and the indent is doing the work of showing that 'this is a separate element / thing'.

It's a small point, though :) Do what makes you happiest ::)
